distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/cometd/CometDClient.java

```diff
@@ -177,8 +179,46 @@ public void subscribe(String channel) {
     }
 
     /**
-     * Gets the list of messages received on a given channel.
+     * Returns an action url from the messages of a channel if it contains the search criteria string.
      *
+     * @param searchCriterion a string that will be searched for in the messages.
+     * @param title           the title of the action we want to find.
+     * @return The desired action url if a match for the search criterion is found in the messages.
+     * Otherwise returns an empty string.
+     * @throws Exception
+     */
+    public String findActionUrlBySearchCriterion(String searchCriterion, String title)
+            throws Exception {
+
+        List<String> messages = getAllMessages();
+
+        Optional<String> foundString = messages.stream()
+                .filter(query -> query.contains(searchCriterion))
+                .findAny();
```

Comment:
There's no advantage to using `findAny()` in a sequential search of the stream. If you were to run a parallelized search, it would potentially be faster. 

As this is a sequential search, it's more intuitive to explicitly call `findFirst()`.